pkg/controller/operconfig/operconfig_controller.go

```diff
@@ -335,6 +335,13 @@ func (r *ReconcileOperConfig) Reconcile(ctx context.Context, request reconcile.R
 	allResources = append(allResources, deployments...)
 	r.podReconciler.SetResources(allResources)
 
+	// Remove ovnkube-node object if ovn db route doesn't exist in hypershift management cluster
+	if bootstrapResult.OVN.OVNKubernetesConfig.HypershiftConfig.Enabled && bootstrapResult.OVN.OVNKubernetesConfig.HypershiftConfig.Route == "" {
+		log.Printf("Removing ovnkube-node objects from rendered objs")
+		objs = apply.RemoveObject(objs, "ovnkube-node", "DaemonSet")
```

Comment:
This logic should be in `pkg/render` IMHO. Then we need to return an additional "Progressing" status from `Render()`. Which we should have been doing all along, especially with the logic to wait for node rollout on upgrades...

Reply:
I updated the PR with the following changes:
1) add create-wait annotation (similar to create-only) to indicate that the ovnkube-node object needs to wait
2) return a progressing boolean value in render* all the way back to operator-controller
3) set/unset the progressing condition based on the returned value from render*.

Is this what you have in mind?

I wonder if we can just check the existense of create-wait annotation to set/unset the progressing status.